src/managers/TileManager.ts

```diff
@@ -12,43 +11,44 @@ import { TileState } from "definitions/enums";
 import { EventEmitter } from "utilities/EventEmitter";
 import PlayerManager from "./PlayerManager";
 import { Player } from "entities/Player";
-import { api } from "../utilities/api";
 import SessionManager from "./SessionManager";
-import { AdjacencyMap } from "utilities/AdjacencyMap";
 
 class TileManager {
     public readonly onSync: EventEmitter;
     private list: Nullable<Tile[]>;
-    private adjacencyMap: Nullable<AdjacencyMap>;
-    private connectionsMap: Map<int, int[]>;
 
     public constructor() {
         this.onSync = new EventEmitter();
         this.list = null;
-        this.adjacencyMap = null;
-        this.connectionsMap = this.createConnectionsMap();
-    }
-
-    public getConnectionsMap(): Map<int, int[]> {
-        return this.connectionsMap;
     }
 
     public getAll(): Nullable<Tile[]> {
         return this.list;
     }
-    public getAdjacencyMap(): Nullable<AdjacencyMap> {
-        return this.adjacencyMap;
+
+    public getAllExceptGoal(): Tile[] {
+        const placedTiles: Nullable<Tile[]> = this.getPlaced();
+        assert(placedTiles);
+        const preplacedTiles: Tile[] = this.getPreplaced();
+        const startingTile: Tile[] = preplacedTiles.filter(
+            (tile: Tile) => tile.coordinateX === 0,
+        );
+        return [...startingTile, ...placedTiles];
     }
 
-    public updateAdjacencyMap(): void {
-        this.adjacencyMap = new AdjacencyMap(this.getRelevantInWorld());
+    public getAllInWorld(): Tile[] {
+        const prePlaced: Tile[] = this.getPreplaced();
+        const placed: Nullable<Tile[]> = this.getPlaced();
+        assert(placed);
+        return [...prePlaced, ...placed];
     }
 
-    private getRelevantInWorld(): Tile[] {
-        const placedTiles: Nullable<Tile[]> = this.getPlaced();
-        assert(placedTiles);
-        const startingTile: Tile[] = this.getStarting();
-        return [...placedTiles, ...startingTile];
+    public getGoal(): Tile[] {
```

Comment:
I imagine that getGoal are the vein tiles... Wouldn't it be more intuitive to call it getVein? just because we always talked about veins and not goals. What do you think?

Reply:
great suggestion